src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -2093,6 +2092,36 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             }
         }
 
+#nullable enable
+        public override BoundNode VisitCall(BoundCall node)
+        {
+            BoundExpression? receiverOpt = node.ReceiverOpt;
+
+            if (node.Method.RequiresInstanceReceiver)
+            {
+                receiverOpt = (BoundExpression?)this.Visit(receiverOpt);
+            }
+            else
+            {
```

Comment:
It looks like this is necessary because `this.Visit` is unable to handle a `BoundTypeExpression`. Is that right?

Reply:
> It looks like this is necessary because this.Visit is unable to handle a BoundTypeExpression. Is that right?

In genera it can do that. However, since the first pass specially visits BoundCall, we have to mirror that here as well in order to make sure that the count of nodes is in sync across passes.